File: global_safety_enterprises/global_safety_enterprises/report/today's_follow_ups/today's_follow_ups.py
# Copyright (c) 2023, Thirvusoft and contributors
# For license information, please see license.txt

import frappe

def execute(filters = None):

	columns = get_columns(filters)

	data = get_data(filters)

	return columns, data

def get_columns(filters):

	columns = [

		{
			'fieldname': 'lead_quotation_id',
			'fieldtype': 'Data',
			'label': 'Lead / Quotation ID',
			'width': 182
		},

		{
			'fieldname': 'lead_name',
			'fieldtype': 'Data',
			'label': 'Lead Name',
			'width': 182
		},

		{
			'fieldname': 'lead_owner',
			'fieldtype': 'Data',
			'label': 'Lead Owner',
			'width': 182
		},

		{
			'fieldname': 'territory',
			'fieldtype': 'Link',
			'label': 'Territory',
			'options': 'Territory',
			'width': 182
		},

		{
			'fieldname': 'status',
			'fieldtype': 'Data',
			'label': 'Status',
			'width': 182
		},

		{
			'fieldname': 'contact_number',
			'fieldtype': 'Data',
			'label': 'Contact Number',
			'width': 182
		},

		{
			'fieldname': 'remarks',
			'fieldtype': 'Data',
			'label': 'Remarks',
			'width': 400
		},

		{
			'fieldname':'description',
			'fieldtype':'Small Text',
			'label':'Description',
			'width':400
		},
	]

	return columns

def get_data(filters):

	data = [{}]

	return data

File: global_safety_enterprises/global_safety_enterprises/report/today's_follow_ups/today's_follow_ups.js
// Copyright (c) 2023, Thirvusoft and contributors
// For license information, please see license.txt

frappe.query_reports["Today's Follow Ups"] = {
	"filters": [
		{
			fieldname: 'date',
			label: 'Date',
			fieldtype: 'Date',
			default: 'Today',
			reqd: 1
		},
		{
			fieldname: 'user',
			label: 'Follow Up By',
			fieldtype: 'Autocomplete',
			options: []
		},
		{
			fieldname: 'lead',
			label: 'Lead',
			fieldtype: 'Check',
			default: 1
		},
		{
			fieldname: 'quotation',
			label: 'Quotation',
			fieldtype: 'Check',
			default: 1
		}
	],

	onload: function(report){

		frappe.db.get_value("User", {"name": frappe.session.user}, "username", (r) => {
			frappe.query_report.set_filter_value('user', r.username);
		})

		frappe.call({

			method: "global_safety_enterprises.global_safety_enterprises.report.today's_follow_ups.today's_follow_ups.get_user_list",

			args: {user: frappe.session.user},

			callback(r){

				if ((r.message).length < 2){
					frappe.query_report.page.fields_dict.user.df.hidden = 1;
				}
				
				frappe.query_report.page.fields_dict.user.set_data(r.message);
				frappe.query_report.page.fields_dict.user.refresh();
			}
		})

	}
};